Modules/binascii.c

```diff
@@ -1293,9 +1265,9 @@ binascii_a2b_qp_impl(PyObject *module, Py_buffer *data, int header)
                       (ascii_data[in+1] >= 'a' && ascii_data[in+1] <= 'f') ||
                       (ascii_data[in+1] >= '0' && ascii_data[in+1] <= '9'))) {
                 /* hexval */
-                ch = hexval(ascii_data[in]) << 4;
+                ch = _PyLong_DigitValue[ascii_data[in]] << 4;
```

Comment:
What if replace the above checks with `_PyLong_DigitValue[ascii_data[in]] < 16 && _PyLong_DigitValue[ascii_data[in+1]] < 16` (with saving intermediate results in variables)?

Reply:
Do you mean like here? https://github.com/python/cpython/pull/4586/commits/bf57e77ac4ddae79fa52a028eacbc533a0e4e622#diff-37791d6fab7c332f1ef6b333d28b7aceR1261

Reply:
Yes, exactly. What were the problems with this?

Reply:
It becomes slower than it was for some reason.
Before:
```
$ ./python -m timeit -s "from binascii import a2b_qp; b = b'=AF'*2**20" "a2b_qp(b)"
100 loops, best of 5: 3.39 msec per loop
```
After:
```
$ ./python -m timeit -s "from binascii import a2b_qp; b = b'=AF'*2**20" "a2b_qp(b)"
100 loops, best of 5: 3.72 msec per loop
```